bukkit/src/main/java/me/william278/husksync/bukkit/util/PlayerSetter.java

```diff
@@ -260,6 +278,48 @@ private static void setPlayerPotionEffects(Player player, PotionEffect[] effects
         }
     }
 
+    private static void nativeSyncPlayerAdvancements(final Player player, final List<DataSerializer.AdvancementRecord> advancementRecords) {
+        final Object playerAdvancements = AdvancementUtils.getPlayerAdvancements(player);
+
+        // Clear
+        AdvancementUtils.clearPlayerAdvancementsMap(playerAdvancements);
+
+        final Map<Object, List<String>> syncAdvancementMap = new HashMap<>();
+        advancementRecords.forEach(advancementRecord -> {
+            NamespacedKey namespacedKey = Objects.requireNonNull(
+                    NamespacedKey.fromString(advancementRecord.advancementKey()),
+                    "Invalid Namespaced key of " + advancementRecord.advancementKey()
+            );
+
+            Advancement bukkitAdvancement = Bukkit.getAdvancement(namespacedKey);
+            if (bukkitAdvancement == null) {
+                // todo: write logging
+                return;
+            }
+
+            syncAdvancementMap.put(
+                    AdvancementUtils.getHandle(bukkitAdvancement),
+                    advancementRecord.awardedAdvancementCriteria()
+            );
+        });
+
+        // todo: sync date of get advancement
```

Comment:
Do you still plan on doing this?

Reply:
Why not? I think this is what plugin users expect to see

Reply:
I was just wondering as there's still a TODO there :)